gatsby-browser.js

```diff
@@ -4,4 +4,29 @@
  * See: https://www.gatsbyjs.org/docs/browser-apis/
  */
 
-// You can delete this file if you're not using it
+import React from "react"
+import { checkSession } from "./src/utils/auth"
+
+const Loader = () => <pre>Loading Session ...</pre>
+
+class SessionCheck extends React.Component {
+  state = {
+    loading: true,
+  }
+
+  componentWillMount() {
+    checkSession(this.handleCheckSession)
+  }
+
+  handleCheckSession = () => {
+    this.setState({ loading: false })
+  }
+
+  render() {
+    return this.state.loading ? <Loader /> : <>{this.props.children}</>
+  }
+}
+
+export const wrapRootElement = ({ element }) => {
+  return <SessionCheck>{element}</SessionCheck>
+}
```

Comment:
### 8. {Refresh user session on every visit}

> Lines {12} - {32}

Our `SessionCheck` component triggers the `checkSession` function in auth.js to renew the session using an Auth0 cookie. By using the `gatsby-browser.js` `wrapRootElement` api we can do this when the app is first mounted and before the DOM has been rendered.

[Next →](https://github.com/erikdstock/gatsby-starter-ts-auth0/pull/2#pullrequestreview-239278342)